app/Services/FetchWorkflowJobsService.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use App\DTO\WorkflowRunDTO;
+use App\Exceptions\FetchingWorkflowJobsErrorException;
+use App\Http\Integrations\GithubConnector;
+use App\Http\Integrations\Requests\GetWorkflowJobsRequest;
+use App\Models\Organization;
+use App\Models\Repository;
+use App\Models\User;
+use App\Models\WorkflowJob;
+use Exception;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Validation\UnauthorizedException;
+
+class FetchWorkflowJobsService
+{
+    public function __construct(
+        protected GithubConnector $githubConnector,
+    ) {}
+
+    public function fetchWorkflowJobs(WorkflowRunDTO $workflowRunDto): void
+    {
+        $repository = Repository::query()
+            ->where("id", $workflowRunDto->repositoryId)
+            ->firstOrFail();
+
+        $organization = Organization::query()
+            ->where("id", $repository->organization_id)
+            ->firstOrFail();
+
+        $user = User::query()->where("id", Auth::user()->id)->firstOrFail();
```

Comment:
Is it needed?

Reply:
I think so

Reply:
`Auth::user()` method getting user data too, and in that line you are doing twice the same thing.